BardMusicPlayer.Catalog/CatalogManager.cs

```diff
@@ -0,0 +1,359 @@
+﻿using BardMusicPlayer.Notate.Objects;
+using LiteDB;
+using System;
+using System.Collections.Generic;
+
+namespace BardMusicPlayer.Catalog
+{
+    public sealed class CatalogManager : IDisposable
+    {
+        private readonly LiteDatabase dbi;
+        private bool disposedValue;
+
+        /// <summary>
+        /// Internal constructor; this object is constructed with a factory pattern.
+        /// </summary>
+        /// <param name="dbi"></param>
+        private CatalogManager(LiteDatabase dbi)
+        {
+            this.dbi = dbi;
+            this.disposedValue = false;
+        }
+
+        /// <summary>
+        /// Create a new instance of the CatalogManager based on the given LiteDB database.
+        /// </summary>
+        /// <param name="dbPath"></param>
+        /// <returns></returns>
+        public static CatalogManager CreateInstance(string dbPath)
```

Comment:
this should be a static path. we want to abstract as much from the user as possible. if they want to specify a path in settings, then cool. but this still remains private.

Reply:
private path* so, i.e. they don't have to create an instance. the definition of a singleton is never requiring a constructor. just allow the user to set the song path, which is probably also a no go.

Reply:
we might need to discuss how to merge directories given from users into the database. that's how i would want it to work.